### PR TITLE
Individual model reports

### PR DESCRIPTION
**Pull request by kathsherratt**
I'm attempting to combine code from the ensemble report and the evaluation reports to create individual model reports. 

At the moment I keep running into strange errors which appear when sourcing the script to `compile-models-report.R`, but have no trouble when knitting interactively (from `model-report.Rmd`). The latest error is:
```
 In filter(., !is.na(forecast_date)) :
  internal error -3 in R_decompress1
```

I've been trying different solutions - clearing the cache each time, changing parameters - but no luck... 

I'll continue trying but thought I'd open a draft PR in case anyone has time to have a look as well.

### CODE REVIEW

File: code/reports/rmdchunks/model-scores.Rmd [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<!--- Table comparing the performance of each model. The table is followed by
+an explanation text detailing the meaning of each column/index. --->
+
+## Predictive performance {.tabset .tabset-fade}
+
+```{r ranking, results='asis'}
+cat("Scores separated by target and forecast horizon.")
+if (!("model" %in% names(params ))) {
```
Comment (Bisaloo) [inline code/reports/rmdchunks/model-scores.Rmd:8]:
```suggestion
if (is.null(params$model)) {
```

File: code/reports/rmdchunks/model-ranking-table.Rmd [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+if (nrow(model_df) > 0) {
+  cat("The table shows the skill of the ", params$model,
+      " model (`Model skill`), the ensemble model (`Ensemble skill`), and",
+      " the best individual model (`Best skill`).\n")
+
+  top_ranked <- df %>%
+    arrange(location_name, skill) %>%
+    group_by(location_name) %>%
+    slice(1) %>%
```
Comment (Bisaloo) [inline code/reports/rmdchunks/model-ranking-table.Rmd:33]:
```suggestion
    group_by(location_name) %>%
    slice(which.min(skill)) %>%
```

File: code/reports/rmdchunks/model-ranking-table.Rmd [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  df <- df %>%
+    rename(skill = rel_wis)
+  model_df <- model_df %>%
+    rename(skill = rel_wis)
+}
+
+if (nrow(model_df) > 0) {
+  cat("The table shows the skill of the ", params$model,
+      " model (`Model skill`), the ensemble model (`Ensemble skill`), and",
+      " the best individual model (`Best skill`).\n")
```
Comment (Bisaloo) [inline code/reports/rmdchunks/model-ranking-table.Rmd:28]:
```suggestion
  cat("The table shows the skill of the ", params$model,
      "model (`Model skill`), the ensemble model (`Ensemble skill`), and",
      "the best individual model (`Best skill`).\n")
```

File: code/reports/rmdchunks/model-ranking-table.Rmd [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  filter(target_variable == target_variables[variable],
+         horizon == this_horizon)
+
+model_df <- df %>%
+  filter(model == params$model)
+
+if (all(is.na(model_df$rel_wis))) {
+  cat("Skill is shown as relative absolute error; weighted interval score",
+      " is not calculated as the 23 quantile levels required are not being",
+      " provided by the model.")
```
Comment (Bisaloo) [inline code/reports/rmdchunks/model-ranking-table.Rmd:12]:
```suggestion
  cat("Skill is shown as relative absolute error; weighted interval score",
      "is not calculated as the 23 quantile levels required are not being",
      "provided by the model.")
```

File: code/reports/rmdchunks/model-wis-time.Rmd [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+         target_variable == target_variables[variable],
+         horizon == this_horizon)
+
+model_df <- df %>%
+  filter(model == params$model)
+
+if (all(is.na(model_df$wis))) {
+  cat("Skill is shown as weighted interval score",
+      " is not calculated as the 23 quantile levels required are not being",
+      " provided by the model.")
```
Comment (Bisaloo) [inline code/reports/rmdchunks/model-wis-time.Rmd:14]:
```suggestion
  cat("Skill is shown as weighted interval score",
      "is not calculated as the 23 quantile levels required are not being",
      "provided by the model.")
```

File: code/reports/rmdchunks/model-ranking-plots.Rmd [hunk 6]
```diff
@@ -0,0 +1,42 @@
+```{r, results = 'asis'}
+library(forcats)
+
+lookup_variables <- names(target_variables)
+names(lookup_variables) <- target_variables
+skill_table <- table %>%
+  mutate(variable = lookup_variables[target_variable])
```
Comment (Bisaloo) [inline code/reports/rmdchunks/model-ranking-plots.Rmd:7]:
```suggestion
skill_table <- table %>%
  mutate(variable = fct_recode(target_variable, !!!target_variables))
```

Comment (sbfnk) [reply code/reports/rmdchunks/model-ranking-plots.Rmd:7]: nice - we'll need to add a `forcats` dependency, right?

File: code/reports/rmdchunks/model-ranking-plots.Rmd [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+## select best individual models
+selected <- skill_table %>%
+  ## FIXME: do we need a better way to identify hub models?
```
[the remark this answers is not in the document]
Comment (sbfnk) [reply code/reports/rmdchunks/model-ranking-plots.Rmd:11]: see https://github.com/epiforecasts/covid19-forecast-hub-europe/issues/1054

File: code/reports/rmdchunks/model-ranking-table.Rmd [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+    arrange(location_name, skill) %>%
+    group_by(location_name) %>%
+    slice(1) %>%
+    ungroup() %>%
+    select(location_name, `Best skill` = skill)
+
+  ## if not doing the ensemble, show the ensemble
+  if (!grepl("hub-ensemble$", params$model)) {
+    ensemble <- df %>%
+      filter(grepl("hub-ensemble$", model)) %>%
```
[the remark this answers is not in the document]
Comment (sbfnk) [reply code/reports/rmdchunks/model-ranking-table.Rmd:40]: https://github.com/epiforecasts/covid19-forecast-hub-europe/issues/1054

File: code/reports/rmdchunks/filter-forecasts.Rmd [hunk 9]
```diff
@@ -19,6 +24,11 @@ data <- data %>%
   filter(horizon %in% horizons)
 table <- table %>%
   filter(horizon %in% horizons)
+
+if (!is.null(params$model)) {
+  data <- data %>%
+    filter(model == {{model}})
```
Comment (Bisaloo) [inline code/reports/rmdchunks/filter-forecasts.Rmd:30]:
```suggestion
    filter(model == params$model)
```
sorry, I must have forgotten to save this edit but I think this is even clearer

**Pull request merged**